### ISSUE TITLE
Querying multiple fields referencing the same underlying relation gives incorrect outputs

### ISSUE THREAD

**Issue by rsomani95** [MrThearMan/graphene-django-query-optimizer#87]
### I have read the docs thoroughly before making this bug report.

- [X] Yes

### I have read through other open issues, and my issue is not a duplicate.

- [X] Yes

### What version of the library you are using?

0.4.1

### Which python version are you using?

3.12

### What operating system are you on?

Mac

### Description

When querying multiple fields that reference the same underlying relation, the `developers_alt` field doesn't respect its scope. It returns all `developers` from all `housing_companies`.

```python
class HousingCompanyNode(IsTypeOfProxyPatch, DjangoObjectType):
    developers = DjangoConnectionField(DeveloperNode)
    developers_alt = DjangoConnectionField(DeveloperNode, field_name="developers")

    class Meta:
        model = HousingCompanyProxy
        interfaces = (relay.Node,)
        connection_class = CustomConnection
        filterset_class = HousingCompanyFilterSet
```

```graphql
{
  pagedHousingCompanies {
    edges {
      node {
        developersAlt {
          totalCount
          edges {
            node {
              id
              name
            }
          }
        }
        developers {
          totalCount
          edges {
            node {
              id
              name
            }
          }
        }
      }
    }
  }
}
```

Returned data for the first `housingCompany` node:

```json
{
  "data": {
    "pagedHousingCompanies": {
      "edges": [
        {
          "node": {
            "developersAlt": {
              "totalCount": 10,
              "edges": [
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZTo5",
                    "name": "April Brown"
                  }
                },
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZToy",
                    "name": "Carol Horn"
                  }
                },
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZToxMA==",
                    "name": "Carrie Diaz"
                  }
                },
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZTo2",
                    "name": "Douglas Cameron"
                  }
                },
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZTo0",
                    "name": "Kevin Wolfe"
                  }
                },
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZTo3",
                    "name": "Melissa Sullivan"
                  }
                },
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZTo4",
                    "name": "Miguel Waters"
                  }
                },
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZToz",
                    "name": "Natalie James"
                  }
                },
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZTo1",
                    "name": "Robert Hall"
                  }
                },
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZTox",
                    "name": "Ruth Jones"
                  }
                }
              ]
            },
            "developers": {
              "totalCount": 1,
              "edges": [
                {
                  "node": {
                    "id": "RGV2ZWxvcGVyTm9kZTo2",
                    "name": "Douglas Cameron"
                  }
                }
              ]
            }
          }
        },
```

However, doing the query with only `developersAlt` gives the correct output:

```graphql
{
  pagedHousingCompanies {
    edges {
      node {
        developersAlt {
          totalCount
          edges {
            node {
              id
              name
            }
          }
        }
        # developers {
        #   totalCount
        #   edges {
        #     node {
        #       id
        #       name
        #     }
        #   }
        # }
      }
    }
  }
}
```

**Comment by elyas-hedayat** [comment]
@rsomani95 can you share CustomConnection code  with me?

**Comment by rsomani95** [comment]
@elyas-hedayat [here](https://github.com/MrThearMan/graphene-django-query-optimizer/blob/552a1111079d77e442358eddf74eff13225cf84d/tests/example/types.py#L247-L258)

**Comment by MrThearMan** [comment]
Hey, I decided to rewrite the testing setup for this project over the weekend, which took most of my time, but have made some progress in fixing this in the [fix-alternate-fields](https://github.com/MrThearMan/graphene-django-query-optimizer/tree/fix-alternate-fields) branch. Its not complete yet - there still seems to be some duplication happening between many-to-many entities that are shared between multiple relations.

**Comment by MrThearMan** [comment]
OK, managed to fix this. First, there were some issues with alternate fields not being registered correctly for optimization, since their field clashed with the original field in the optimizer if queried together. Second, the prefetch hack did not account for two prefetches for the same relation occurring in a single query, resulting in duplicate data in nested connections. These should be fixed in [0.4.2](https://github.com/MrThearMan/graphene-django-query-optimizer/releases/tag/v0.4.2).

**Comment by rsomani95** [comment]
@MrThearMan awesome, thanks for the fix. I haven't had a chance to test it out in my project yet, but will let you know if there's any hiccups.
